src/Yarhl/IO/DataStream.cs

```diff
@@ -497,19 +492,102 @@ public void WriteTo(DataStream stream)
                 throw new ObjectDisposedException(nameof(stream));
 
             long currPos = Position;
-            Seek(0, SeekMode.Start);
+            Seek(start, SeekMode.Start);
 
             const int BufferSize = 70 * 1024;
             byte[] buffer = new byte[Length > BufferSize ? BufferSize : Length];
 
-            while (!EndOfStream) {
+            while (!EndOfStream)
```

Comment:
The code convention is to put the braces in the same line :)

Reply:
Done